### PR TITLE
Image API Fix & Makefile

### PR DESCRIPTION
**Pull request by gambol99**
- fixing the typo on the image api action, changing from /image/build -> /build
- updating the base image to alpine 3.6
- making a new cleanup modifications to Makefile

### CODE REVIEW

**Comment by gambol99** [comment]
hi @liron-l ... any chance you can take a look at this one? :-) .. would hate to run off a fork

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,12 @@
+AUTHOR ?= twistlock
+NAME ?= authz-broker
```
Comment (liron-l) [inline Makefile]: If you would like to override this, why not having `IMAGE_NAME ?=`

File: core/route_parser.go [hunk 2]
```diff
@@ -11,7 +11,8 @@ type route struct {
 }
 
 var routes = []route{
-
+	// https://docs.docker.com/reference/api/docker_remote_api_v1.21/#build-image-from-a-dockerfile
+	{pattern: "/build", method: "POST", action: ActionImageBuild},
```
Comment (liron-l) [inline core/route_parser.go]: Can you please add this API in the UT?

File: Dockerfile [hunk 3]
```diff
@@ -1,18 +1,16 @@
-FROM alpine:3.3
+FROM alpine:3.6
```
Comment (liron-l) [inline Dockerfile]: Consider upgrading to 3.7 (it's officially out)

File: core/route_parser.go (hunk 2 above)
Comment (gambol99) [reply core/route_parser.go]: [added](https://github.com/twistlock/authz/pull/48/files#diff-e383ff40d0872136a71e0c9d96b0fafbR17) :-)

File: core/route_parser_test.go [hunk 4]
```diff
@@ -27,40 +29,40 @@ func TestRouteParser(t *testing.T) {
...
-		{"GET", "/v.1.21/networks/id", ActionNetworkInspect},
-		{"POST", "/v.1.21/networks/id/disconnect", ActionNetworkDisconnect},
-		{"POST", "/v.1.21/networks/id/connect", ActionNetworkConnect},
-		{"DELETE", "/v.1.21/networks/id", ActionNetworkRemove},
-		{"DELETE", "/v.1.21/volumes/id", ActionVolumeRemove},
-		{"POST", "/v.1.21/volumes/create", ActionVolumeCreate},
-		{"GET", "/v.1.21/volumes/id", ActionVolumeInspect},
-		{"GET", "/v.1.21/volumes", ActionVolumeList},
-		{"GET", "/v.1.21/images/non_existing", ActionNone},
+		{"POST", "/v1.21/containers/id/attach", ActionContainerAttach},
```
Comment (gambol99) [inline core/route_parser_test.go]: I fixed these up .. the extra dot .. as i'm guessing it was an typo?

**Comment by gambol99** [comment]
hi @liron-l  ... i've made the changes you requested .. One thing I changes was the VERSION on the [Makefile](https://github.com/twistlock/authz/pull/48/files#diff-b67911656ef5d18c4ae36cb6741b7965R3) .. which i figured it would be nice to just change from the cli version?? .. but retain the override. If prefer not, just say and i'll revert though :-)

**Comment by gambol99** [comment]
fixed a quick typo at in [Makefile](https://github.com/twistlock/authz/pull/48/files#diff-b67911656ef5d18c4ae36cb6741b7965R3) .. needed a $$ no $ on the awk command .. :-)

**Comment by liron-l** [comment]
Thanks @gambol99, I'm not sure parsing the code to fetch the version is considered optional.
You can do everything directly in the makefile, specifically:
1. Define the version in the make file
2. Use ldflag to push the version to the binary as part of go build. [see](https://stackoverflow.com/questions/11354518/golang-application-auto-build-versioning):
```
```--ldflags "-X \"locationtogetversion.Version=$(version)\" ```
```
If the steps above are not clear, just make the version static in the makefile (that is, not related to the binary), and I will push your changes.

**Comment by gambol99** [comment]
hi @liron-l .. like [this](https://github.com/twistlock/authz/pull/48/commits/feba6276afd2de358babddb934afe7f0b84add4e) i.e. take the version from the Makefile passing via the flags .. note I keep the image separate but default to app version, purely to retain a 'latest' image ... If you happy, i'll rebase and squash the commits

**Comment by liron-l** [comment]
Sure @gambol99 just set the default version to something reasonable (e.g., `version = "1.0.0"`).

**Comment by gambol99** [comment]
hi @liron-l .. done :-)

**Comment by liron-l** [comment]
Thanks @gambol99

**Pull request merged**